presto-sqlserver/src/main/java/com/facebook/presto/plugin/sqlserver/SqlServerClient.java

```diff
@@ -18,49 +18,148 @@
+            .put(REAL, "float(24)")
+            .put(VARBINARY, "varbinary")
+            .put(DATE, "date")
+            .put(TIME, "time")
+            .put(TIMESTAMP, "datetime2")
+            .put(TIMESTAMP_WITH_TIME_ZONE, "datetimeoffset")
+            .build();
+
     @Inject
     public SqlServerClient(JdbcConnectorId connectorId, BaseJdbcConfig config)
             throws SQLException
     {
         super(connectorId, config, "\"", new SQLServerDriver());
+        this.connectionProperties.setProperty("initializationString", "SET QUOTED_IDENTIFIER, ANSI_NULLS, CONCAT_NULL_YIELDS_NULL, ARITHABORT ON;");
+        this.connectionProperties.setProperty("sendStringParametersAsUnicode", "false");
+        this.connectionProperties.setProperty("applicationName", "Presto");
+    }
+
+    @Override
+    public Set<String> getSchemaNames()
+    {
+        try (Connection connection = this.driver.connect(this.connectionUrl, this.connectionProperties);
+                ResultSet resultSet = connection.getMetaData().getSchemas()) {
+            ImmutableSet.Builder<String> schemas = ImmutableSet.builder();
+            while (resultSet.next()) {
+                String name = resultSet.getString("TABLE_SCHEM").toLowerCase(ENGLISH);
+                // skip internal schemas
+                if (!"sys".equals(name) && !"information_schema".equals(name)) {
+                    schemas.add(name);
+                }
+            }
+            return schemas.build();
+        } catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
     }
 
     @Override
     public void commitCreateTable(JdbcOutputTableHandle handle)
     {
         StringBuilder sql = new StringBuilder()
                 .append("sp_rename ")
-                .append(singleQuote(handle.getCatalogName(), handle.getSchemaName(), handle.getTemporaryTableName()))
+                .append("'" + quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTemporaryTableName()) + "'")
                 .append(", ")
-                .append(singleQuote(handle.getTableName()));
+                .append("'" + handle.getTableName() + "'");
 
-        try (Connection connection = getConnection(handle)) {
-            execute(connection, sql.toString());
+        try (Connection connection = this.getConnection(handle)) {
+            this.execute(connection, sql.toString());
         }
         catch (SQLException e) {
             throw new PrestoException(JDBC_ERROR, e);
         }
     }
 
-    private static String singleQuote(String catalog, String schema, String table)
+    @Override
+    public PreparedStatement getPreparedStatement(Connection connection, String sql)
+            throws SQLException
     {
-        return singleQuote(catalog + "." + schema + "." + table);
+        connection.setAutoCommit(false);
+        PreparedStatement statement = connection.prepareStatement(sql);
+        statement.setFetchSize(1000);
```

Comment:
It's from the PostgreSQL plugin, in this case it'll default to 128 which is a bit small for large data sets.